tests/test_launcher.py

```diff
@@ -84,9 +85,9 @@ def test_gpu_launch_arg_additional_arg(monkeypatch):
 
 
 def test_kwargs():
-    with pytest.raises(RuntimeError):
+    with pytest.raises(LauncherError.MeshingModeError):
         pyfluent.launch_fluent(abc=1, meshing_mode=True)
-    with pytest.raises(TypeError):
+    with pytest.raises(LauncherError.UnexpectedKeywordArgumentError):
         pyfluent.launch_fluent(abc=1, xyz=2)
```

Comment:
@hpohekar It's kind of remarkable that the exception types are different. That's the great thing about this kind of testing. Please continue to add more scenarios, and as you go, you can review if the raised exceptions match the error scenarios (and update the the tests and implementations as necessary). Thanks.

Reply:
Yes sure.

Reply:
@hpohekar we should keep the comment unresolved while the points are still being addressed. Let me specify what they are. I was referring to the arbitrary exception type differences. The original `launch_fluent` code here was wrong. Both exceptions should have been `TypeError`, and that has led you to create distinct types, which looks very odd in this test, right? We should ask about the value of adding new exception types in each case. These `TypeError`s will crop up do to incorrect usage of the interface. `TypeError` is already adequate for that, and the message contains all the context. In creating new exception types, we should focus on the more sophisticated failure scenarios.

Reply:
@seanpearsonuk 

`The original launch_fluent code here was wrong. Both exceptions should have been TypeError, and that has led you to create distinct types, which looks very odd in this test, right?` 

Even if we had corrected both of them to `TypeError` still we need to create and inherit distinctly as we have done in `TransactionError`.

`We should ask about the value of adding new exception types in each case.`

The issue description says that we don't want lower-level built-in exception types, therefore we are updating all existing exceptions. We have not decided any criteria to add new exception types (places where we want to add new exceptions types and places where we want to use built-in exception types).

Reply:
We can replace `LauncherError.MeshingModeError` with `LauncherError.UnexpectedKeywordArgumentError`.

We can also remove `LauncherError.MeshingModeError`.

Reply:
Replaced `LauncherError.MeshingModeError` with `LauncherError.UnexpectedKeywordArgumentError`